<?xml version="1.0"?>

<launch>

    <!-- Associate to the robot description parameter, the urdf file that model the robot-->
    <param name="robot_description" command = "$(find xacro)/xacro $(find trajectory_pkg)/urdf/ur5_traj.xacro" />

    <!-- Set the position in empty world-->
    <arg name="x" default="0" />
    <arg name="y" default="0" />
    <arg name="z" default="1.015" />
    <!-- <arg name="z" default="0" /> -->
    <arg name="R" default="0" />
    <arg name="P" default="0" />
    <arg name="Y" default="0" />

    <!-- put world file as argument-->
    <arg name="world_file" default = "$(find trajectory_pkg)/worlds/my_scene.world" />

    <!-- Launch Gezabo world in paused-->
    <include file = "$(find gazebo_ros)/launch/empty_world.launch" >
        <arg name="use_sim_time" value="true"/>
        <arg name="debug" value="false"/>
        <arg name="gui" value="true"/>
        <arg name="paused" value="true" />
        <arg name="world_name" value="$(arg world_file)"/>
    </include>  

<!-- ************************************************************************ -->

    <!-- Spawn the robot using the package gazebo_ros-->
    <node name="spawn_the_robot" pkg="gazebo_ros" type="spawn_model"  output="screen" 
        args="-urdf -param robot_description -model ur5 -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg R) -P $(arg P) -Y $(arg Y)"/>


    <!-- Load the controllers-->
    <rosparam command="load" file="$(find trajectory_pkg)/config/ur5_effort_controller.yaml"/>

    <!-- Load the node controller spawner-->
    <node name="controller_spawner" pkg="controller_manager" type="spawner"  
        output="screen" respawn="false" ns="/ur5" args="joint_state_controller trajectory_controller">        
    </node>    


    <!-- Convert joint states to TF transform for rviz and rqt-->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"  
        output="screen" respawn="false">
        
    </node>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <remap from="/joint_states" to="/ur5/joint_states"/> 


  </launch>




  